appinventor/components/src/com/google/appinventor/components/runtime/ListView.java

```diff
@@ -922,6 +930,91 @@ public void ListData(String propertyValue) {
     setAdapterData();
   }
 
+  /**
+   * Sets whether the ListView has line separator or not.
+   *
+   * @param hasSeparator boolean value to determine the ListView has separator or not
+   */
+  @DesignerProperty(editorType = PropertyTypeConstants.PROPERTY_TYPE_BOOLEAN, defaultValue = "False")
+  @SimpleProperty(userVisible = false)
+  public void HasSeparator(boolean hasSeparator)  {
+    this.hasSeparator = hasSeparator;
+    setSeparator();
+  }
+
+  /**
+   * Returns whether the ListView has line separator or not.
+   *
+   * @return the boolean value represent the ListView has separator or not
+   */
+  @SimpleProperty(userVisible = false)
+  public boolean HasSeparator() {
+    return this.hasSeparator;
+  }
+
+  /**
+   * Sets the color of the ListView's line separator if ListView has line separator.
+   *
+   * @param separatorColor the int value of line separator color
+   */
+  @DesignerProperty(editorType = PropertyTypeConstants.PROPERTY_TYPE_COLOR, defaultValue = Component.DEFAULT_VALUE_COLOR_PINK)
+  @SimpleProperty(userVisible = false)
+  public void SeparatorColor(int separatorColor)  {
+    this.separatorColor = separatorColor;
+    setSeparator();
+  }
+
+  /**
+   * Returns the color of the ListView's line separator if ListView has line separator.
+   *
+   * @return the int value of line separator color
+   */
+  @SimpleProperty(userVisible = false)
+  public int SeparatorColor() {
+    return this.separatorColor;
+  }
+
+  /**
+   * Sets the size of the ListView's line separator it would be a height of the line separator
+   * in the case of vertical orientation and width of the line separator in the case of horizontal orientation.
+   *
+   * @param separatorSize the int value of line separator size
+   */
+  @DesignerProperty(editorType = PropertyTypeConstants.PROPERTY_TYPE_NON_NEGATIVE_INTEGER, defaultValue = "1")
+  @SimpleProperty(userVisible = false)
+  public void SeparatorSize(int separatorSize)  {
+    this.separatorSize = separatorSize;
+    setSeparator();
+  }
+
+  /**
+   * Sets the size of the ListView's line separator it would be a height of the line separator
+   * in the case of vertical orientation and width of the line separator in the case of horizontal orientation.
+   *
+   * @return the int value of line separator size
+   */
+  @SimpleProperty(userVisible = false)
+  public int SeparatorSize()  {
+    return this.separatorSize;
+  }
+
+  private void setSeparator() {
+    if(hasSeparator)  {
+      GradientDrawable gradientDrawable = new GradientDrawable(GradientDrawable.Orientation.BOTTOM_TOP, new int[]{separatorColor, separatorColor});
+      DividerItemDecoration dividerItemDecoration;
+      if(orientation == ComponentConstants.LAYOUT_ORIENTATION_VERTICAL) {
```

Comment:
When Orientation is set to horizontal, I see a vertical separator, which seems correct. However, I also see a horizontal separator below the listview element that is set to default size and color.